### PR TITLE
Improve `cleanup-repo-filelist-actions`’s reliability

### PR DESCRIPTION
**Pull request by kidonng**
1. LINKED ISSUES:
   https://github.com/sindresorhus/refined-github/issues/3527#issuecomment-687519926

2. TEST URLS:
   - Repo root: https://github.com/sindresorhus/refined-github
   - Repo tree: https://github.com/sindresorhus/refined-github/tree/master/source
   - File: https://github.com/sindresorhus/refined-github/blob/master/source/options.css

3. SCREENSHOT:
   None

### CODE REVIEW

**Comment by fregante** [comment]
Is this done?

**Comment by fregante** [comment]
This will also need a `rgh-clean-actions` class as described in #3094

**Comment by kidonng** [comment]
Waiting for #3646

**Comment by yakov116** [comment]
I get an error here

https://github.com/sindresorhus/refined-github/blob/master/.editorconfig
```js
Uncaught TypeError: Cannot read property 'classList' of null
    at add (refined-github.js:1626)
    at runAdd (refined-github.js:17277)
    at applyChanges (refined-github.js:17193)
    at SelectorObserver.addRootNodes (refined-github.js:17713)
    at Array.processBatchQueue (refined-github.js:17134)
    at MutationObserver.handleMutations (refined-github.js:17168)
```

**Comment by kidonng** [comment]
I get that error without this PR too

Looking into it

**Comment by kidonng** [comment]
Caused by https://github.com/sindresorhus/refined-github/pull/3646/files#diff-5ba5120afe524f2d5eed502b044996a1db29ba24221b41edee322940746bf553R20

Add button doesn't exist on files page

**Comment by yakov116** [comment]
That's why we had 3 approvals 🤣.

**Comment by yakov116** [comment]
That's why we had 3 approvals 🤣.

**Comment by kidonng** [comment]
Another issue:

image

I was going to take a look at that PR but then it got merged... 🐛

**Comment by kidonng** [comment]
I would also like to take this opportunity to rename this feature to `clean-repo-filelist-actions` since we have many `clean-*` features.

**Comment by yakov116** [comment]
> I would also like to take this opportunity to rename this feature to `clean-repo-filelist-actions` since we have many `clean-*` features.

Please do https://github.com/sindresorhus/refined-github/pull/3514#discussion_r496379255

**Comment by kidonng** [comment]
I also turned the "Download" text of `download-folder-button` into a tooltip, as mentioned in https://github.com/sindresorhus/refined-github/pull/3405#issue-457099585

image

image

File: source/refined-github.ts [hunk 1]
```diff
@@ -185,7 +185,7 @@ import './features/keyboard-navigation';
 import './features/vertical-front-matter';
 import './features/use-first-commit-message-for-new-prs';
 import './features/linkify-user-edit-history-popup';
-import './features/cleanup-repo-filelist-actions';
+import './features/clean-repo-filelist-actions';
```
Comment (fregante) [inline source/refined-github.ts:188]: Migration needed in `background.ts`

Comment (kidonng) [reply source/refined-github.ts:188]: Did you mean `options-storage.ts`?

File: source/options-storage.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ const defaults = Object.assign({
 const migrations = [
 	featureWasRenamed('sticky-conversation-sidebar', 'sticky-sidebar'), // Merged in September
```
Comment (yakov116) [inline source/options-storage.ts:21]:
```suggestion
```
W can drop this line

Comment (fregante) [reply source/options-storage.ts:21]: It’s still October 😃 include it in your November lint

**Comment by yakov116** [comment]
```js
refined-github.js:1627 Uncaught TypeError: Cannot read property 'replaceWith' of null
    at add (refined-github.js:1627)
    at runAdd (refined-github.js:17277)
    at applyChanges (refined-github.js:17193)
    at SelectorObserver.addRootNodes (refined-github.js:17713)
    at Array.processBatchQueue (refined-github.js:17134)
    at MutationObserver.handleMutations (refined-github.js:17168)
```

On https://github.com/sindresorhus/refined-github/tree/master/source

File: source/features/clean-repo-filelist-actions.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+				addButtonText.classList.replace('d-md-flex', 'd-md-block');
+				addButtonText.firstChild!.replaceWith(<PlusIcon/>);
+			}
+		}
+	});
+
+	observe('get-repo summary:not(.rgh-clean-actions)', {
+		add(button) {
+			button.classList.add('tooltipped', 'tooltipped-ne', 'rgh-clean-actions');
```
Comment (ocarreterom) [inline source/features/clean-repo-filelist-actions.tsx:33]: Could you check if _click outside to close_ works?. That's the reason I moved that classes from `<summary>` to `<details>` [here](https://github.com/sindresorhus/refined-github/commit/b501d71a83857a7068d8e8c1ca5ef21bd7a93233#diff-5ba5120afe524f2d5eed502b044996a1db29ba24221b41edee322940746bf553R17).

File: source/features/clean-repo-filelist-actions.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		add(searchButton) {
+			searchButton.classList.add('tooltipped', 'tooltipped-ne', 'rgh-clean-actions');
+			searchButton.setAttribute('aria-label', 'Go to file');
+
+			searchButton.firstChild!.replaceWith(<SearchIcon/>);
+
+			// Exclude logged out, mobile or file pages
+			const addButtonText = searchButton.nextElementSibling?.querySelector('.d-md-flex');
+			if (addButtonText) {
+				addButtonText.parentElement!.classList.add('tooltipped', 'tooltipped-ne');
```
Comment (ocarreterom) [inline source/features/clean-repo-filelist-actions.tsx:22]: 👇 (if `parentElement` is a `<summary>`)

Comment (fregante) [reply source/features/clean-repo-filelist-actions.tsx:22]: In what case it's not a summary? My understanding is that it's a summary when it's a dropdown, and it always is.

File: readme.md [hunk 5]
```diff
@@ -129,7 +129,7 @@ Thanks for contributing! 🦋🙌
 - [](# "go-to-action-from-file") [Adds a link to access the past runs of a GitHub Action workflow when seeing the workflow configuration file.](https://user-images.githubusercontent.com/1402241/80146153-ab6d6400-85b1-11ea-9f38-e87950692a62.png)
 - [](# "clone-branch") [Clone a branch from the branches list.](https://user-images.githubusercontent.com/16872793/76802029-2a020500-67ad-11ea-95dc-bee1b1352976.png)
 - [](# "fork-source-link-same-view") [Points the “Forked from user/repository” link to current folder or file in the upstream repository.](https://user-images.githubusercontent.com/1402241/84795784-3722d000-aff8-11ea-9b34-97c01acf4fd4.png)
-- [](# "cleanup-repo-filelist-actions") [Replaces the labels of some simple buttons on repository filelists with icons, making them take less space.](https://user-images.githubusercontent.com/44045911/88551471-7a3f7c80-d055-11ea-82f1-c558b7871824.png)
+- [](# "clean-repo-filelist-actions") [Replaces the labels of some simple buttons on repository filelists with icons, making them take less space.](https://user-images.githubusercontent.com/44045911/96283396-bbb94580-100e-11eb-9480-17954bb3a811.png)
```
Comment (fregante) [inline readme.md:132]: Some padding is ok, but this is excessive. Can you avoid it and instead use higher-resolution screenshots? Zoom the page at 200% if possible before taking a screenshot

**Comment by fregante** [comment]
There are 2 bugs related to this feature, can you ensure this PR fixes them and include them as `Fixes #123 and fixes #456`?

https://github.com/sindresorhus/refined-github/issues/3685
https://github.com/sindresorhus/refined-github/issues/3675

**Comment by fregante** [comment]
Closing this since it’s inactive to allow anyone to possibly work on it. Feel free to open a new PR based on this code.

**Pull request closed without merging**